packages/nimble-components/src/list-option-group/index.ts

```diff
@@ -0,0 +1,157 @@
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import {
+    observable,
+    attr,
+    volatile,
+    Observable
+} from '@microsoft/fast-element';
+import { styles } from './styles';
+import { template } from './template';
+import { ListOption } from '../list-option';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-list-option-group': ListOptionGroup;
+    }
+}
+
+/**
+ * A nimble-styled HTML listbox option group
+ */
+export class ListOptionGroup extends FoundationElement {
+    /**
+     * The label for the group.
+     *
+     * @public
+     * @remarks
+     * If a label is also provided via slotted content, the label attribute
+     * will have precedence.
+     */
+    @attr
+    public label?: string;
+
+    /**
+     * The hidden state of the element.
+     *
+     * @public
+     * @defaultValue - false
+     * @remarks
+     * HTML Attribute: hidden
+     */
+    @attr({ mode: 'boolean' })
+    public override hidden = false;
+
+    /**
+     * @internal
+     * This attribute is required to allow use-cases that offer dynamic filtering
+     * (like the Select) to visually hide groups that are filtered out, but still
+     * allow users to use the native 'hidden' attribute without it being affected
+     * by the filtering process.
+     */
+    @attr({ attribute: 'visually-hidden', mode: 'boolean' })
+    public visuallyHidden = false;
+
+    /**
+     * @internal
+     */
+    @attr({ attribute: 'top-separator-visible', mode: 'boolean' })
+    public topSeparatorVisible = false;
+
+    /**
+     * @internal
+     */
+    @attr({ attribute: 'bottom-separator-visible', mode: 'boolean' })
+    public bottomSeparatorVisible = false;
+
+    /** @internal */
+    @observable
+    public hasOverflow = false;
+
+    /** @internal */
+    public labelSlot!: HTMLSlotElement;
+
+    /** @internal */
+    @observable
+    public listOptions!: ListOption[];
+
+    /** @internal */
+    @volatile
+    public get labelContent(): string {
+        if (this.label) {
+            return this.label;
+        }
+
+        if (!this.$fastController.isConnected) {
+            return '';
+        }
+
+        const nodes = this.labelSlot.assignedNodes();
+        return nodes
+            .filter(node => node.textContent?.trim() !== '')
+            .map(node => node.textContent?.trim())
+            .join(' ');
+    }
+
+    private readonly hiddenOptions: Map<ListOption, boolean> = new Map();
```

Comment:
🤦‍♂️...yes, it should be a `Set`. I tracked the hidden options instead of the visible options mainly because I preferred this line of code for readability:
```
this.visuallyHidden = this.hiddenOptions.size === this.listOptions.length;
```
I'm fine changing it if you have a stronger preference.